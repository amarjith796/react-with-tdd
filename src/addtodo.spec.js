import React from 'react';
import { shallow } from 'enzyme';
import AddTodo from './Todo/addtodo'
describe("AddTodo", () => {

    let wrapper;
    beforeEach(() => wrapper = shallow(<AddTodo handleAddTask={jest.fn()} editText={''} handleUpdateTodo={jest.fn()} />));
    it('should render correctly', () => expect(wrapper).toMatchSnapshot());
    it('should render a <div/>', () => {
        expect(wrapper.find("div").length).toBeGreaterThanOrEqual(1);
    })
})