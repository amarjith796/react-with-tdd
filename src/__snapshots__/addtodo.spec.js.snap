// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTodo
    handleAddTask={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          autoFocus={true}
          onKeyDown={[Function]}
          placeholder="Add a Todo Task"
          style={
            Object {
              "flex": 1,
            }
          }
        />,
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "steelblue",
              "border": "1px solid steelblue",
              "cursor": "pointer",
              "padding": "10px",
            }
          }
        >
          Addtodo
        </button>,
      ],
      "style": Object {
        "display": "flex",
        "width": "60vw",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": true,
          "onKeyDown": [Function],
          "placeholder": "Add a Todo Task",
          "style": Object {
            "flex": 1,
          },
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Addtodo",
          "onClick": [Function],
          "style": Object {
            "background": "steelblue",
            "border": "1px solid steelblue",
            "cursor": "pointer",
            "padding": "10px",
          },
        },
        "ref": null,
        "rendered": "Addtodo",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            autoFocus={true}
            onKeyDown={[Function]}
            placeholder="Add a Todo Task"
            style={
              Object {
                "flex": 1,
              }
            }
          />,
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "steelblue",
                "border": "1px solid steelblue",
                "cursor": "pointer",
                "padding": "10px",
              }
            }
          >
            Addtodo
          </button>,
        ],
        "style": Object {
          "display": "flex",
          "width": "60vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": true,
            "onKeyDown": [Function],
            "placeholder": "Add a Todo Task",
            "style": Object {
              "flex": 1,
            },
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Addtodo",
            "onClick": [Function],
            "style": Object {
              "background": "steelblue",
              "border": "1px solid steelblue",
              "cursor": "pointer",
              "padding": "10px",
            },
          },
          "ref": null,
          "rendered": "Addtodo",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
